File: src/main/java/team4/footwithme/team/service/TeamMemberService.java
package team4.footwithme.team.service;

import team4.footwithme.member.domain.Member;
import team4.footwithme.team.service.request.TeamMemberServiceRequest;
import team4.footwithme.team.service.response.TeamResponse;

import java.util.List;

public interface TeamMemberService {
    List<TeamResponse> addTeamMembers(Long teamId, TeamMemberServiceRequest request);

    List<TeamResponse> getTeamMembers(Long teamId);

    Long deleteTeamMemberByCreator(Long teamId, Long teamMemberId, Member member);

    Long deleteTeamMember(Long teamId, Member member);
}

File: src/main/java/team4/footwithme/team/service/TeamMemberService.kt
package team4.footwithme.team.service

import team4.footwithme.member.domain.Member
import team4.footwithme.team.service.request.TeamMemberServiceRequest
import team4.footwithme.team.service.response.TeamResponse

interface TeamMemberService {
    fun addTeamMembers(teamId: Long, request: TeamMemberServiceRequest?): List<TeamResponse>

    fun getTeamMembers(teamId: Long): List<TeamResponse>

    fun deleteTeamMemberByCreator(teamId: Long, teamMemberId: Long, member: Member?): Long

    fun deleteTeamMember(teamId: Long, member: Member?): Long?
}
